plugins/modules/one_vnet.py

```diff
@@ -0,0 +1,434 @@
+        return self.get_template(lambda template: (template.ID == template_id))
+
+    def get_template_by_name(self, name):
+        return self.get_template(lambda template: (template.NAME == name))
+
+    def get_template_instance(self, requested_id, requested_name):
+        if requested_id:
+            return self.get_template_by_id(requested_id)
+        else:
+            return self.get_template_by_name(requested_name)
+
+    def get_networks_ar_pool(self, template):
+        ar_pool = []
+        for ar in template.AR_POOL:
+            ar_pool.append({
+                # These params will always be present
+                'ar_id': ar['AR_ID'],
+                'mac': ar['MAC'],
+                'size': ar['SIZE'],
+                'type': ar['TYPE'],
+                # These are optional so firstly check for presence
+                # and if not present set value to Null
+                'allocated': getattr(ar, 'ALLOCATED', 'Null'),
+                'ip': getattr(ar, 'IP', 'Null'),
+                'global_prefix': getattr(ar, 'GLOBAL_PREFIX', 'Null'),
+                'parent_network_ar_id': getattr(ar, 'PARENT_NETWORK_AR_ID', 'Null'),
+                'ula_prefix': getattr(ar, 'ULA_PREFIX', 'Null'),
+                'vn_mad': getattr(ar, 'VN_MAD', 'Null'),
+            })
+        return ar_pool
+
+    def get_template_info(self, template):
+        info = {
+            'id': template.ID,
+            'name': template.NAME,
+            'template': template.TEMPLATE,
+            'user_name': template.UNAME,
+            'user_id': template.UID,
+            'group_name': template.GNAME,
+            'group_id': template.GID,
+            'permissions': {
+                'owner_u': template.PERMISSIONS.OWNER_U,
+                'owner_m': template.PERMISSIONS.OWNER_M,
+                'owner_a': template.PERMISSIONS.OWNER_A,
+                'group_u': template.PERMISSIONS.GROUP_U,
+                'group_m': template.PERMISSIONS.GROUP_M,
+                'group_a': template.PERMISSIONS.GROUP_A,
+                'other_u': template.PERMISSIONS.OTHER_U,
+                'other_m': template.PERMISSIONS.OTHER_M,
+                'other_a': template.PERMISSIONS.OTHER_A
+            },
+            'clusters': template.CLUSTERS.ID,
+            'bridge': template.BRIDGE,
+            'bride_type': template.BRIDGE_TYPE,
+            'parent_network_id': template.PARENT_NETWORK_ID,
+            'vm_mad': template.VM_MAD,
+            'phydev': template.PHYDEV,
+            'vlan_id': template.VLAN_ID,
+            'outer_vlan_id': template.OUTER_VLAN_ID,
+            'used_leases': template.USED_LEASES,
+            'vrouters': template.VROUTERS.ID,
+            'ar_pool': self.get_networks_ar_pool(template)
+        }
+
+        return info
+
+    def create_template(self, name, template_data):
+        if not self.module.check_mode:
+            self.one.vn.allocate("NAME = \"" + name + "\"\n" + template_data)
```

Comment:
It would probably make sense to properly escape `name`, like when it contains double quotes or newlines.